splice_machine/src/test/java/com/splicemachine/derby/test/framework/SpliceUnitTest.java

```diff
@@ -1018,27 +1031,85 @@ public static int getNumberOfFiles(String dirPath)
         }
     }
 
     /// execute sql query 'sqlText' and expect a certain formatted result
-    public static void SqlExpectToString( SpliceWatcher methodWatcher, String sqlText, String expected, boolean sort ) throws Exception
+    public static void sqlExpectToString(SpliceWatcher methodWatcher, String sqlText, String expected, boolean sort ) throws Exception
     {
         try( ResultSet rs = methodWatcher.executeQuery(sqlText) ) {
             String val = sort
                     ? TestUtils.FormattedResult.ResultFactory.toString(rs)
                     : TestUtils.FormattedResult.ResultFactory.toStringUnsorted(rs);
-            Assert.assertEquals(expected, val);
+            Assert.assertEquals(sqlText, expected, val);
+        }
+    }
+
+
+    /**
+     * check if there's any transactions still open
+     * if getOldestActiveTransaction is two times the same in a row, there's a stale transaction
+     * otherwise, we will get every time different ids since also the CALL SYSUTIL_... will open a mini-transaction
+     *
+     * Note that this is only correct if the test ends and nothing else is running anymore
+     * (e.g. @Category({SerialTest.class}) ). Because of this, the default should be failOnError = false.
+     * @param failOnError if true, fail on error (don't do this on parallel tests). otherwise, just LOG.
+     */
+    public static void waitForStaleTransactions(SpliceWatcher methodWatcher, String name, int numSeconds,
+                                                boolean failOnError) throws Exception
+    {
+        Logger LOG = Logger.getLogger("SpliceUnitTest");
+        LOG.info("checking for stale transactions");
+
+        int oldest1 = getOldestActiveTransaction(methodWatcher);
+        for(int i=0; i < numSeconds; i++) {
+            int oldest2 = getOldestActiveTransaction(methodWatcher);
+            if( oldest1 != oldest2 )
+                return;
+            Thread.sleep(1000);
+            oldest1 = oldest2;
+        }
+        if( failOnError ) {
+            Assert.fail(name + " failed to close all transactions.");
         }
+        else {
+            // if you see this error, this is a hint that something might be left open, especially if you see
+            // multiple of these messages. turn failOnError=true and check tests one by one.
+            LOG.info("WARNING: " + name + " failed to close all transactions. This might be due to multiple " +
+                "tests running in parallel.");
+        }
+    }
+
+    public static int getOldestActiveTransaction(SpliceWatcher methodWatcher) throws SQLException {
+        try (ResultSet rs = methodWatcher.executeQuery("call SYSCS_UTIL.SYSCS_GET_OLDEST_ACTIVE_TRANSACTION()"))
+        {
+            Assert.assertEquals(true, rs.next());
+            return rs.getInt(1);
+        }
+    }
+
+    @AfterClass
+    public static void waitForStaleTransactions() throws Exception {
```

Comment:
This adds 5 seconds per class, at 311 ITs that's 25 more minutes of runtime. Maybe we should disable this by default and only enable it if VacuumIT is failing to debug?

Reply:
@martinrupp explained on the normal case this shouldn't wait at all